core/state/statedb.go

```diff
@@ -726,3 +728,43 @@ func (db *DB) IsValidator(addr common.Address) bool {
 	}
 	return so.IsValidator(db.db)
 }
+
+// AddReward distributes the reward to all the delegators based on stake percentage.
+func (db *DB) AddReward(validator common.Address, reward *big.Int) error {
```

Comment:
so if this fails, do I early return as well at call site?

Reply:
Yes, but the caller should really check the existence of the validator too.